Slow Subscribers: clarify datagram behavior, add cross-references (#27)
englishm/moq-relay-dos: draft-englishm-moq-relay-dos.md

(see {{Section 9.4 of MOQT}}).
Without such information,
a subscriber that consumes data more slowly than it is produced
will cause data to accumulate at the relay.
That memory pressure could affect other subscribers
on the same relay
(see {{flow-control-limitations}} and {{resource-isolation}}).
This can happen because the subscriber's network is slow,
because the subscriber's application isn't consuming data
fast enough, or because the subscriber is deliberately
not consuming data at all.

Objects sent on QUIC streams are reliably delivered,
so the relay's only mechanism to bound buffering
is to terminate the subscription entirely
by resetting the stream.
The relay cannot selectively drop individual objects
while keeping the subscription active,
which means any resource bound requires full termination —
creating a denial-of-service risk
where the only defense disrupts service.
Objects sent as datagrams can simply be dropped
when a buffer forms,
which naturally limits accumulation
but does not eliminate the processing cost
of receiving and discarding them.

A relay also cannot apply QUIC flow control backpressure
to the upstream publisher on behalf of one slow subscriber

> without affecting all other subscribers on that track.

Author: ```suggestion
without affecting other downstream subscribers of that track.
```

causes data to accumulate at the relay.

> The relay must buffer objects waiting to be read,

Reviewer: nit: I know this is existing text but whenever using lowercase words like "must" people might be triggered. Is the act of buffering is a consequence or a requirement? In the prior sentence it sounds like the accumlation is a consequence but that's perhaps due to a requirement on MoQT?

If MoQT does have requirements, we might want to reconstruct the para to something like:

Due to the relay delivery requirements in Section X of MoQT,  a subscriber that consumes data more slowly than it is produced will cause the data to accumulate in the relay, which could create memory pressure that subsequently affects the relay's ability to service other subscribers (see {{flow-control-limitations}} and {{resource-isolation}}).

and that memory pressure could affect other subscribers
on the same relay
(see {{flow-control-limitations}} and {{resource-isolation}}).
This can happen because the subscriber's network is slow,
because the subscriber's application isn't consuming data
fast enough, or because the subscriber is deliberately
not consuming data at all.

Objects sent on QUIC streams are reliably delivered,
so a slow subscriber causes unbounded buffering

> unless the relay intervenes.

Reviewer: It mght help to explain or cite why bounding the buffering is not possible. Otherwise I'm thinking why not? can't a relay impose some local policy? in my HTTP servers I have all sorts of implementation-specific limits (in size or time dimensions) and if those are breached I blow away the request/response. For example, if a client doesn't let me send response body bytes in a timely manner, eventujally I reset the stream

Reviewer: also not 100% clear to me if the relay can exert backpressue to the producer or not (I'd guess no, since one slow client shouldn't affect serving other subsribers)
